.github/workflows/repackage-native-libs.yml

```diff
@@ -0,0 +1,183 @@
+name: repackage-native-libs
+
+on:
+  push:
+    tags:
+      - v*
+
+
+jobs:
+
+  ci_wf:
+    uses: AsamK/signal-cli/.github/workflows/ci.yml@master
+      # ${{ github.repository }} not accpeted here
+
+
+  lib_to_jar:
+    needs: ci_wf
+    runs-on: ubuntu-latest
+
+    outputs:
+      signal_cli_version: ${{ steps.cli_ver.outputs.signal_cli_version }}
+      release_id: ${{ steps.create_release.outputs.id }}
+
+    steps:
+
+      - name: Download signal-cli build from CI workflow
+        uses: actions/download-artifact@v2
+
+      - name: Get signal-cli version
+        id: cli_ver
+        run: |
+          #echo ${GITHUB_REF#refs/tag/}
+          tree .
+          mv ./*/*.tar.gz .
+          ver=$(ls ./*.tar.gz | xargs basename | sed -E 's/signal-cli-(.*).tar.gz/\1/')
+          echo $ver
+          echo "::set-output name=signal_cli_version::${ver}"
+          tar -xzf ./*.tar.gz
+
+      - name: Get signal-client jar version
+        id: lib_ver
+        run: |
+          JAR_PREFIX=signal-client-java-
+          jar_file=$(find ./signal-cli-*/lib/ -name "$JAR_PREFIX*.jar")
+          jar_version=$(echo "$jar_file" | xargs basename | sed "s/$JAR_PREFIX//; s/.jar//")
+          echo "$jar_version"
+          echo "::set-output name=signal_client_version::$jar_version"
+
+      - name: Download signal-client builds
+        env:
+          RELEASES_URL: https://github.com/signalapp/libsignal-client/releases/download/
+          FILE_NAMES: signal_jni.dll libsignal_jni.dylib
+          SIGNAL_CLIENT_VER: ${{ steps.lib_ver.outputs.signal_client_version }}
+        run: |
+          for file_name in $FILE_NAMES; do
+            curl -sOL "${RELEASES_URL}/v${SIGNAL_CLIENT_VER}/${file_name}"  # note: added v
+          done
+          tree .
+
+      - name: Replace Windows lib
+        env:
+          SIGNAL_CLI_VER: ${{ steps.cli_ver.outputs.signal_cli_version }}
+          SIGNAL_CLIENT_VER: ${{ steps.lib_ver.outputs.signal_client_version }}
+        run: |
+          mv signal_jni.dll libsignal_jni.so
+          zip -u ./signal-cli-${SIGNAL_CLI_VER}/lib/signal-client-java-${SIGNAL_CLIENT_VER}.jar  ./libsignal_jni.so
+          tar -czf signal-cli-${SIGNAL_CLI_VER}-Windows.tar.gz signal-cli-${SIGNAL_CLI_VER}/
```

Comment:
Maybe it would be better to use a zip file for windows? I think there's no built-in support for tar.gz files.

Reply:
[Turns out](https://superuser.com/a/1431799), Win 10 now has a built-in `tar` accessible from `cmd.exe`. Also, my impression is, everyone on windows has a 7-Zip (or some similar 3rd party program) installed anyway. Not to mention, the signal-cli releases so far have been packaged as `.tar.gz`, and windows users have not been complaining :) 
But let me know, and I can change it to `.zip` if needed.